### ISSUE TITLE
Fix S2094 FP: Implicit parameterless constructor widens the scope of the base class constructor

### ISSUE THREAD

**Issue by rjgotten** [SonarSource/sonar-dotnet#7591]
### Description

S2094 regarding an empty class can be raised incorrectly in situations where an implicit parameterless constructor widens the scope modifier of the base class constructor. E.g. where an inherited class publishes a previously protected parameterless constructor.

### Repro steps

```cs
public class Foo
{
  protected Foo() { }
}

public class Bar : Foo
{
}
```

### Expected behavior

S2094 is not raised where the implicit parameterless constructor widens the scope modifier of the base class parameterless constructor.

### Actual behavior

S2094 is raised.

### Known workarounds

Explicitly implement the constructor. E.g.

```cs
public class Foo
{
  protected Foo() { }
}

public class Bar : Foo
{
  public Bar() : base() { }
}
```

However, this (correctly!) raises S3253 regarding having defined a redundant constructor.
In effect, the only way to fix this is to suppress S2094.

### Related information

* C#/VB.NET Plugins version : C# 9.4.0.72892 (SonarLint  7.0.0.74072)
* Visual Studio version: 17.6.5
* MSBuild / dotnet version: dotnet 7.0.306
* Operating System: Windows 11

**Comment by zsolt-kolbay-sonarsource** [comment]
Hi @rjgotten. Thank you for reporting this. Confirmed as False Positive.
